server/src/main/java/com/objectcomputing/checkins/services/file/FileServicesImpl.java

```diff
@@ -0,0 +1,226 @@
+                CheckIn checkIn = checkInServices.read(checkInID);
+                validate(checkIn == null, "Unable to find checkin record with id %s", checkInID);
+
+                Set<CheckinDocument> checkinDocuments = checkinDocumentServices.read(checkInID);
+                for (CheckinDocument cd : checkinDocuments) {
+                    File file = drive.files().get(cd.getUploadDocId()).execute();
+                    result.add(setFileInfo(file, cd));
+                }
+            }
+
+            return HttpResponse.ok(result);
+        } catch (GoogleJsonResponseException e) {
+            LOG.error("Error occurred while retrieving files from Google Drive.", e);
+            return HttpResponse
+                    .status(HttpStatus.valueOf(e.getStatusCode()))
+                    .body(e.getContent());
+        } catch (IOException e) {
+            LOG.error("Error occurred while retrieving files from Google Drive.", e);
+            return HttpResponse.status(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @Override
+    public HttpResponse<?> downloadFiles(@NotNull String uploadDocId) {
+
+        try {
+            OutputStream outputStream = new ByteArrayOutputStream();
+            java.io.File file = java.io.File.createTempFile("tmp", ".txt");
+            file.deleteOnExit();
+            FileWriter myWriter = new FileWriter(file);
+
+            Drive drive = googleDriveAccessor.accessGoogleDrive();
+            validate(drive == null, "Unable to access Google Drive");
+
+            drive.files().get(uploadDocId).executeMediaAndDownloadTo(outputStream);
+            myWriter.write(String.valueOf(outputStream));
+            myWriter.close();
+
+            return HttpResponse.status(HttpStatus.OK).body(file);
+        }  catch (HttpResponseException e) {
+            LOG.error("Error occurred while retrieving files from Google Drive.", e);
+            return HttpResponse
+                    .status(HttpStatus.valueOf(e.getStatusCode()))
+                    .body(e.getContent());
+        } catch (IOException e) {
+            LOG.error("Error occurred while retrieving files from Google Drive.", e);
+            return HttpResponse.status(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @Override
+    public HttpResponse<FileInfoDTO> uploadFile(@NotNull UUID checkInID, @NotNull CompletedFileUpload file) {
+
+        CheckIn checkIn = checkInServices.read(checkInID);
+        validate(checkIn == null, "Unable to find checkin record with id %s", checkInID);
+        validate((file.getFilename() == null || file.getFilename().equals("")), "Please select a file before uploading.");
+
+        // create folder name in the format name-date
+        String subjectName = memberProfileServices.getById(checkIn.getTeamMemberId()).getName();
+        String sb = subjectName.concat(LocalDate.now().toString());
+        sb = sb.replaceAll("\\s", "");
+        final String directoryName = sb;
```

Comment:
Can we make this name match the format in the DoD? it's ok to include time too...but it should at least start: `lastname, firstname`. I know it's kind of a pain.

Reply:
We don't have an option to fetch last and first names from memberprofile. When creating a new user, we are getting 'name' attribute from auth object and saving it in memberprofile. This attribute includes first and last names of the user.

In the above code, directoryName will be something like - PramukhBagur2020-10-08
If a user has a middle name / middle initial, it will be something like - PramukhDayanandaBagur2020-10-08

We can skip time since only 1 directory will be created for the user on a day. DOD -
- [ ] endpoint creates a folder within the configured root location named lastname, firstname - YYYY.MM.DD if it does not already exist

When the user tries to upload second document on the same day, new directory won't be created.

Could you suggest a workaround for doing `lastname, firstname`?